### ISSUE TITLE
aggregating responses based on response's field

### ISSUE THREAD

**Issue by alioygur** [devopsfaith/krakend#155]
Hello, can I do this?

I have a `hotels` endpoint hotel response have destination_id field. Can I get hotel with its destination by one request?

`/hotels/{id}`

```json
{
    "id": 1,
    "destination_id": 2,
    "name": "x hotel"
}
```

`/destinations/{id}`

```
{
    "id": 1,
    "name": "Germany"
}
```

**Comment by kpacha** [comment]
Hi, @alioygur 

in order to optimize the QoS, conditional requests are not supported out of the box.

Anyway, there are 2 ways to get the feature:

- using a custom martial modifier to decode the initial response, trigger and process the secondary request.
- adding a wrapper over the `proxy. NewMergeDataMiddleware` so it triggers the received proxy functions sequentially

cheers!

**Comment by alioygur** [comment]
Thank for your response

Daniel López <notifications@github.com>, 1 Eki 2018 Pzt, 18:50 tarihinde
şunu yazdı:

**Comment by kpacha** [comment]
@alioygur , I pushed a branch with a quick'n'dirty implementation. Please, check it out

https://github.com/devopsfaith/krakend/commit/7142ae43d417ab8cf2a6c61c8d27f7a79c7c5f1b

This is the basic configuration for the endpoint of your example:

```{
    "endpoint": "/hotels/{id}",
    "backend": [
        {
            "host": [
                "https://xxx.xxx"
            ],
            "url_pattern": "/hotels/{id}"
        },
        {
            "host": [
                "https://yyy.xxx"
            ],
            "url_pattern": "/destinations/{resp0_destinations_id}"
        }
    ],
    "extra_config": {
        "github.com/devopsfaith/krakend/proxy": {
            "sequential": true
        }
    }
},```

I think adding the feature will not hurt the framework once the implementation is a little more mature and (well) tested.

cheers!

**Comment by alioygur** [comment]
@kpacha this is a great job! thank you. I will try.

**Comment by kpacha** [comment]
@alioygur  I just merged the PR into the 0.7.0-dev branch, so it will included into the next release

**Comment by kpacha** [comment]
released
